netdata/netdata
# chore(cgroups.plugin): improve cgroups name resolution in k8s

##### Summary

This PR adds the following improvements:

- do not query K8s API for [pause](https://github.com/kubernetes/kubernetes/tree/master/build/pause) containers (a container that holds the network namespace for the pod).
- cache (store on the host filesystem) `/api/v1/pods` response and check it before querying K8s API (only for containers).
- simplifies renaming code.

---

##### Test Plan

- K8s (requires adding debug logging messages)
  - check pause container identification logic.
  - check that cache works (hit/miss).
- run N docker containers, check name resolution works.

##### Additional Information


<details> <summary>For users: How does this change affect me?</summary>
  
</details>
